core/src/snapshot_utils.rs

```diff
@@ -260,35 +228,37 @@ pub fn untar_snapshot_in<P: AsRef<Path>, Q: AsRef<Path>>(
 
 fn rebuild_bank_from_snapshots<P>(
     local_account_paths: String,
-    snapshot_paths: &[SlotSnapshotPaths],
+    unpacked_snapshots_dir: &PathBuf,
     append_vecs_path: P,
 ) -> Result<Bank>
 where
     P: AsRef<Path>,
 {
-    // Rebuild the last root bank
-    let last_root_paths = snapshot_paths
-        .last()
+    let mut snapshot_paths = get_snapshot_paths(&unpacked_snapshots_dir);
+    assert!(snapshot_paths.len() <= 1);
```

Comment:
We shouldn't assert here right?  It's not an internal bug if the length of snapshot_paths is <= 1, it's just a bad snapshot that the user told us to load.  Instead return an `Err`

Reply:
Yes that's better. I'll update it.